app/models/service_template_transformation_plan_task.rb

```diff
@@ -129,4 +201,60 @@ def create_error_status_task(userid, msg)
       :message => msg
     )
   end
+
+  def conversion_options_source_provider_vmwarews_vddk(_storage)
+    {
+      :vm_name            => source.name,
+      :transport_method   => 'vddk',
+      :vmware_fingerprint => source.host.thumbprint_sha1,
+      :vmware_uri         => URI::Generic.build(
+        :scheme   => 'esx',
+        :userinfo => CGI.escape(source.host.authentication_userid),
+        :host     => source.host.ipaddress,
+        :path     => '/',
+        :query    => { :no_verify => 1 }.to_query
+      ).to_s,
+      :vmware_password    => source.host.authentication_password
+    }
+  end
+
+  def conversion_options_source_provider_vmwarews_ssh(storage)
+    {
+      :vm_name          => URI::Generic.build(:scheme => 'ssh', :userinfo => 'root', :host => source.host.ipaddress, :path => "/vmfs/volumes").to_s + "/#{storage.name}/#{source.location}",
+      :transport_method => 'ssh'
+    }
+  end
+
+  def conversion_options_destination_provider_rhevm(cluster, storage)
+    {
+      :rhv_url             => URI::Generic.build(:scheme => 'https', :host => destination_ems.hostname, :path => '/ovirt-engine/api').to_s,
+      :rhv_cluster         => cluster.name,
+      :rhv_storage         => storage.name,
+      :rhv_password        => destination_ems.authentication_password,
+      :install_drivers     => true,
+      :insecure_connection => true
+    }
+  end
+
+  def conversion_options_destination_provider_openstack(cluster, storage)
+    {
+      :osp_environment            => {
+        :os_no_cache         => true,
+        :os_auth_url         => URI::Generic.build(
+          :scheme => destination_ems.security_protocol == 'non-ssl' ? 'http' : 'https',
+          :host   => destination_ems.hostname,
+          :port   => destination_ems.port,
+          :path   => destination_ems.api_version
```

Comment:
Not needed for this PR but seeing this twice (once for vmware and once for openstack) I wonder if we should put this as a generic method on the endpoint so you could do something like `dest_ems.default_endpoint.uri`